src/integrationtest/java/com.microsoft.aad.msal4j/ClientCredentialsIT.java

```diff
@@ -72,8 +72,14 @@ private void assertAcquireTokenCommon(String clientId, IClientCredential credent
 
         Assert.assertNotNull(result);
         Assert.assertNotNull(result.accessToken());
-        // TODO AuthenticationResult should have an getAccountInfo API
-        // Assert.assertEquals(labResponse.getUser().getUpn(), result.getAccountInfo().getUsername());
+
+        result = cca.acquireTokenSilently(SilentParameters
+                .builder(Collections.singleton(KEYVAULT_DEFAULT_SCOPE))
+                .build())
+                .get();
+
+        Assert.assertNotNull(result);
+        Assert.assertNotNull(result.accessToken());
```

Comment:
Might also be worth writing a test that has multiple tokens in cache for same clientId and tenant but different scopes and check that the right token is returned.

Reply:
good point - extended to few scopes and found serious bug